### DISCUSSION TITLE
Mail problem

### DISCUSSION THREAD

**Discussion by BRNSystems** [pocketbase/pocketbase#439] [Q&A]
Hello, I can't use mail in pocketbase. I am running my own smtp server, and pocketbase is using it. Why does this error appear instead of sending an E-mail? 
And pocketbase is behind a reverse proxy via nginx.

Failed to send the test email. Raw error: 
550 5.5.2 <localhost>: Helo command rejected: need fully-qualified hostname.

Can I set the hostname manually somehow?

**Comment by ganigeorgiev** [comment]
This is probably due to some misconfiguration in the local smtp server.

What are you using for the self hosted smtp?

If it is Postfix, to disable the hostname checks you can disable the `reject_invalid_helo_hostname` and `reject_non_fqdn_helo_hostname` [config options](https://www.postfix.org/postconf.5.html).

OR you may also wan to configure the Postfix [`myhostname` paratemer](https://www.postfix.org/BASIC_CONFIGURATION_README.html#myhostname) to a proper fully qualified hostname/domain.

_I'll move the issue to Q&A discussion._

**Reply by BRNSystems** [reply]
Well, myhostname is set to my domain(brn.systems)
And why can't I just change localhost to brn.systems in pocketbase?

**Reply by ganigeorgiev** [reply]
The `550 5.5.2: ...` error is the raw/direct error from the mail server, so I don't think this is a PocketBase configuration issue.

From the Postfix docs in https://www.postfix.org/BASIC_CONFIGURATION_README.html:

> By default, [myhostname](https://www.postfix.org/postconf.5.html#myhostname) is set to the local machine name. If your local machine name is not in fully-qualified domain name form, or if you run Postfix on a virtual interface, **you will have to specify the fully-qualified domain name that the mail system should use.**

**Reply by BRNSystems** [reply]
myhostname = brn.systems
smtp_helo_name = brn.systems
alias_maps = hash:/etc/aliases

**Reply by ganigeorgiev** [reply]
> And why can't I just change localhost to brn.systems in pocketbase?

Hm, but you should be able to do that. I've just tested it locally and the validator work as expected - the Host field must be a valid IP (both v4 and v6) or a valid DNS name. Make sure that there are no typos or spaces before/after the host.

**Reply by BRNSystems** [reply]
Well, I use it for my personal mail, but I think the problem is pocketbase sending localhost instead of FQDN

**Reply by ganigeorgiev** [reply]
What are you using as a `Host` value in the PocketBase Admin UI settings - `brn.systems` or `localhost`?

**Reply by ganigeorgiev** [reply]
@BRNSystems There is a similar issue in the upstream mail library that PocketBase is using - https://github.com/domodwyer/mailyak/issues/60#issuecomment-766111068 (although it is more about the STARTTLS upgrade).

**Reply by BRNSystems** [reply]
where is the host value in pocketbase admin ui

**Reply by BRNSystems** [reply]
If you mean the smtp settings, then yes I set brn.systems as a hostname

**Reply by ganigeorgiev** [reply]
~In the `Settings > Mail settings > "SMTP Server Host"` field.~

> If you mean the smtp settings, then yes I set brn.systems as a hostname

Hm, then I'm not really sure what is causing the issue here. I'll try to find some time this weekend and will test it with a full local postfix setup.

**Reply by BRNSystems** [reply]
OK Thanks

**Reply by BRNSystems** [reply]
Well, after fixing premissions, sendmail works, but I don't think it should send localhost as fqdn when using smtp. And also AFAIK it binds to localhost, so I would welcome a CLI parameter to change hostname, since it is behind a reverse proxy.

**Reply by ganigeorgiev** [reply]
I'll investigate it further with tcpdump but we are ~not sending fixed "localhost" as fqdn~ (update: see below).

> AFAIK it binds to localhost, so I would welcome a CLI parameter to change hostname, since it is behind a reverse proxy.

I'm confused. What do you mean by "parameter to change hostname" - hostname for what? There is already `--http` flag that you can use if you need to change the default server server address and port:

```
./pocketbase serve --http="127.0.0.1:3000"
```

All available flags and comamds could be inspected by running `./pocketbase --help`.

**Reply by ganigeorgiev** [reply]
@BRNSystems Actually maybe you are right that we are sending `localhost` for the EHLO, since that's the default value in `net/smtp` - https://cs.opensource.google/go/go/+/master:src/net/smtp/smtp.go;l=98?q=net%2Fsmtp%20localhost&ss=go%2Fgo.

After the users refactoring in #376 I'll open a PR to the upstream library [mailyak](https://github.com/domodwyer/mailyak) for passing context down to the `net/smtp` instance and will consider including a configuration option that will set an explicit `Hello()` on the smtp client.

**Reply by BRNSystems** [reply]
Thank you very much

**Reply by ganigeorgiev** [reply]
_Just for info in case someone stumble here - support to specify a custom domain for the EHLO/HELO exchange was added in the `develop` branch and it will be available with the next v0.18.0 release (there are no ETAs yet)._
